### ISSUE TITLE
new module FinalFieldCheck

### ISSUE THREAD

**Issue by aclfe** [checkstyle/checkstyle#19512]
I have read check documentation: https://checkstyle.org/checks/xxxxxx/nameofaffectedcheck.html
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**Different style rule says fields should be final by default, checkstyle is missing such a step**. 

```bash
/var/tmp $ javac YOUR_FILE.java
```

/var/tmp $ cat config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="FinalFieldCheck"/>
    </module>
</module>
```

/var/tmp $ cat Example.java
```java
public class Example {
    private final String name = "hello";

    private String email = "hey"; // violation
}

```

**Describe the solution you'd like**
The check would flag regular instance field not static fields.

**Comment by aclfe** [comment]
@romani

**Comment by aclfe** [comment]
on it
